Patch/vinyl

warloque opened the pull request:
Patched with Vinyl instead of now deprecated gulp-util.
Please, update the code base and the dependencies badge)

warloque committed:
Replace gulp-util with gulp4 Vinyl methods

Refactor all gulp-util dependent code as of Gulp v.4 gulp-util is deprecated. 
Add Vinyl virtual file type methods and event streaming instead.

warloque committed:
Upgrade version number. Add and update dependencies.

warloque committed:
Remove gulp-util from dependencies

Add vinyl, upgrade node version. 
* node 0.10–10.6 + npm 6.1.0 + gulp 4 successfully tested

jbdemonte commented on package.json:
```diff
@@ -32,7 +31,7 @@
   },
   "engines": {
-    "node": ">= 0.10"
+    "node": ">= 10.6"
```
could you update it to the LTS at least

warloque replied:
I updated to >=10.6 as it's LTS at the moment. I've found no 10.6LTS-syntax available...

As the LTS concept is one that's controlled by the Node.js LTS Working Group, which is a separate team from the npm CLI, this is likely to be tricky to get into npm. The CLI team isn't going to have time to get to this in the near-to-medium term.

jbdemonte replied:
LTS for version 6 or 8 should be more flexible for users
https://github.com/nodejs/Release#release-schedule

jbdemonte commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "gulp-file-insert",
   "description": "Replace custom tokens by files content",
-  "version": "1.0.3",
+  "version": "1.0.4",
```
update to version 2 plz

jbdemonte commented:
Could you fix the CI plz

Commit:
Fix Buffer usage
Buffer string reader needs `from` method due to current nodejs security recommendations

Commit:
Update CI
Add latest dependecies and remove deprecated ones

Commit:
Update README
- add dependencies' and CI status badges

Commit:
Upgrade NodeJS version in CI

warloque commented:
UPD: updated version, dependencies and travis tests, both have passed OK.

Commit:
Set Nodejs to latest stable version

warloque commented:
UPD: done rollback and CI test passed on node LTS 8.11.3

warloque commented:
Is everything OK? If so, waiting for you accepting the patch)
